internal/http_api/api_response.go

```diff
@@ -91,7 +92,7 @@ func RespondV1(w http.ResponseWriter, code int, data interface{}) {
 
 	if code != 200 {
 		isJSON = true
-		response = []byte(fmt.Sprintf(`{"message":"%s"}`, data))
+		response = []byte(fmt.Sprintf(`{"message":%s}`, strconv.Quote(fmt.Sprintf("%s", data))))
```

Comment:
Since we are building a JSON response, we should just update to use `encoding/json` here so there aren't any escaping issues in the future.

```go
response, _ := json.Marshal(struct {
    Message string `json:"message"`
}{fmt.Sprintf("%s", data)})
```

Reply:
I agree, this is a cleaner and more complete solution. In the case that `json.Marshal` fails, where should the error be propagated? Or is it safe to assume it won't fail in the case?

Reply:
I have updated this. I'm not marking it as resolved until I hear back about the question above.

Reply:
If we were to check for error from `json.Marshal()`, we would log it, and then for the reply probably:

```go
response = []byte(`{"message":"INTERNAL_ERROR"}`)
```

...but it does seem impossible for this `json.Marshal()` to fail.

Reply:
agreed, should be impossible, no need to check for errors

Reply:
OK sounds good :+1: